docs/spec/enums.rst

```diff
@@ -0,0 +1,342 @@
+
+    class Pet(Enum):
+        CAT = 1  # OK
+        DOG: int = 2  # Type checker error
+
+    reveal_type(Pet.CAT)  # Revealed type is Literal[Pet.CAT]
+    reveal_type(Pet.DOG)  # Revealed type is Literal[Pet.DOG]
+
+* Methods, callables, and descriptors (including properties) that are defined
+  in the class are not treated as enum members by the ``EnumType`` metaclass
+  and should likewise not be treated as enum members by a type checker::
+
+    def identity(__x): return __x
+
+    class Pet(Enum):
+        CAT = 1  # Member attribute
+        DOG = 2  # Member attribute
+        
+        converter = lambda __x: str(__x)  # Non-member attribute
+        transform = identity  # Non-member attribute
+
+        @property
+        def species(self) -> str:  # Non-member property
+            return "mammal"
+        
+        def speak(self) -> None:  # Non-member method
+            print("meow" if self is Pet.CAT else "woof")
+
+
+* If using Python 3.11 or newer, the ``enum.member`` and ``enum.nonmember``
+  classes can be used to unambiguously distinguish members from non-members.
+  Type checkers should support these classes::
+
+    class Example(Enum):
+        a = member(1)  # Member attribute
+        b = nonmember(2)  # Non-member attribute
+
+        @member
+        def c(self) -> None:  # Member method
+            pass
+
+    reveal_type(Example.a)  # Revealed type is Literal[Example.a]
+    reveal_type(Example.b)  # Revealed type is int
+    reveal_type(Example.c)  # Revealed type is Literal[Example.c]
+
+
+* An enum class can define a class symbol named ``_ignore_``. This can be a list
+  of names or a string containing a space-delimited list of names that are
+  excluded from the list of members at runtime. Type checkers may support this
+  mechanism::
+
+    class Pet(Enum):
+        CAT = 1  # Member attribute
+        DOG = 2  # Non-member attribute
+        FISH = 3  # Non-member attribute
+        _ignore_ = "DOG FISH"
+
+    reveal_type(Pet.CAT)  # Revealed type is Literal[Pet.CAT]
+    reveal_type(Pet.DOG)  # Revealed type is int (if _ignore_ is supported)
+    reveal_type(Pet.FISH)  # Revealed type is int (if _ignore_ is supported)
+
+
+Member Names
+------------
+
+All enum member objects have an attribute ``_name_`` that contains the member's
+name. They also have a property ``name`` that returns the same name. Type
+checkers may infer a literal type for the name of a member::
+
+    class Color(Enum):
+        RED = 1
+        GREEN = 2
+        BLUE = 3
+
+    reveal_type(Color.RED._name_)  # Revealed type is Literal["RED"] (or str)
+    reveal_type(Color.RED.name)  # Revealed type is Literal["RED"] (or str)
+
+    def func1(red_or_blue: Literal[Color.RED, Color.BLUE]):
+        reveal_type(red_or_blue.name)  # Revealed type is Literal["RED", "BLUE"] (or str)
+
+    def func2(any_color: Color):
+        reveal_type(any_color.name)  # Revealed type is Literal["RED", "BLUE", "GREEN"] (or str)
+
```

Comment:
I'm a complete novice at typing, but it seems to me that the revealed type should be `str`, because it always will be a `str`, and using it directly will never produce an enum member.

Reply:
`Literal["RED", "BLUE", "GREEN"]` means one of the exact strings "RED", "BLUE", or "GREEN", which is what the value would be at runtime. The draft spec allows both this precise type and the more general type `str`.

Reply:
At runtime, it's a `str` value, but it's a specific string. A type checker can determine this statically, so it's better for it to evaluate a string literal type here.